crates/cheatcodes/src/test/expect.rs

```diff
@@ -641,17 +729,125 @@ pub(crate) fn handle_expect_emit(
             return false
         }
 
-        true
+        let expected_count = event_to_fill_or_check.count;
```

Comment:
@grandizzy `is_some_and` and `is_none_or` have the same behaviour here because they always evaluate the predicate i.e `count >= expected_count`, indicating that the entry in the map is always `Some`. 

This is due to the order of checks. We both increment/insert an entry in the map [here](https://github.com/foundry-rs/foundry/blob/7d3a057d500177ed942d7950e000813e681d68e6/crates/cheatcodes/src/test/expect.rs#L699) and cross check with `expected_count` only after we've performed the topics and data checks